app/javascript/spec/vm-server-relationship-form/vm-server-relationship-form.spec.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import fetchMock from 'fetch-mock';
+import VmServerRelationShipForm from '../../components/vm-server-relationship-form';
+import '../helpers/miqAjaxButton';
+
+describe('Flavor form component', () => {
+  const miqAjaxButtonSpy = jest.spyOn(window, 'miqAjaxButton');
+  const servers = {
+    resources: [
+      { href: 'http://localhost:3000/api/servers/7', id: '7', name: 'miq-5.9.0.21-TE4' },
+      { href: 'http://localhost:3000/api/servers/13', id: '13', name: 'miq-5.9.0.21-TE7' },
+      { href: 'http://localhost:3000/api/servers/1', id: '1', name: 'miq-5.9.0.21-TE1' },
+      { href: 'http://localhost:3000/api/servers/12', id: '12', name: 'miq-5.9.0.21-TE11' },
+      { href: 'http://localhost:3000/api/servers/4', id: '4', name: 'miq-5.9.0.21-TE3' },
+      { href: 'http://localhost:3000/api/servers/10', id: '10', name: 'miq-5.9.0.21-TE10' },
+      { href: 'http://localhost:3000/api/servers/14', id: '14', name: 'miq-5.9.0.21-TE13' },
+      { href: 'http://localhost:3000/api/servers/11', id: '11', name: 'miq-5.9.0.21-TE12' },
+      { href: 'http://localhost:3000/api/servers/8', id: '8', name: 'miq-5.9.0.21-TE8' },
+      { href: 'http://localhost:3000/api/servers/2', id: '2', name: 'miq-5.9.0.21-TE2' },
+      { href: 'http://localhost:3000/api/servers/6', id: '6', name: 'miq-5.9.0.21-TE5' },
+      { href: 'http://localhost:3000/api/servers/16', id: '16', name: 'EVM' },
+    ],
+  };
+
+  const props = {
+    vmId: '10',
+  };
+
+  afterEach(() => {
+    fetchMock.reset();
+    miqAjaxButtonSpy.mockReset();
+  });
+
+  it('should render form', () => {
+    const wrapper = mount(<VmServerRelationShipForm {...props} />);
+    expect(toJson(wrapper)).toMatchSnapshot();
```

Comment:
This snapshot does not test the layout after you receive data from server and actually render the form. But i think we don't really need snapshots for data-driven-forms if there are no additional components.